### ISSUE TITLE
eks-prow-build-cluster instability issues as of 2023-06-26

### ISSUE THREAD

**Issue by xmudrii** [kubernetes/k8s.io#5473]
We observed stability issues in the EKS Prow build cluster. Nodes tend to randomly stop working and become NotReady or Ready,SchedulingDisabled.

Instability indicators:

- All nodes are between several hours and 4 days old, but cluster-autoscaler didn't trigger scale down since 2023-06-23
- NotReady nodes are reporting events about blocked tasks
- There were two nodes with the same issue

We checked logs, disk usage, and some other metrics, but everything looks fine for affected nodes. We tried restarting an affected node, the node becomes Ready for a few minutes, but then eventually gets deleted.

Screenshot of events for an affected node:

![image](https://github.com/kubernetes/k8s.io/assets/18719127/5f1eba14-9a34-440b-bab3-ff51b1db4508)

/assign @xmudrii @pkprzekwas 
/sig k8s-infra
/priority critical-urgent

**Comment by ameukam** [comment]
Dow  we have samples for kubelet logs?

**Comment by aojea** [comment]
> Screenshot of events for an affected node:

it seems clear the node hung and lost the connection against the control plane, several process are blocked for more than 121 seconds, we need to undeerstand why they are blocked

**Comment by xmudrii** [comment]
Let's keep a track of kubeadm revert PRs:

- https://github.com/kubernetes/kubeadm/pull/2911
- https://github.com/kubernetes/test-infra/pull/30283
- https://github.com/kubernetes/test-infra/pull/30284

**Comment by tzneal** [comment]
When you try this again, can you run this script on the failed node:

**Comment by xmudrii** [comment]
We have migrated to Bottlerocket Linux on October 11 as recommended by AWS as a way to mitigate this issue. Since then, we had total of three nodes that failed, but in a different way that's not related to this issue. This AMI is known to work:

- AMI ID: ami-0064cd291d1583c3c
- AMI name: bottlerocket-aws-k8s-1.28-x86_64-v1.15.1-264e294c

Short context story: AWS originally recommended migrating to Amazon Linux 2 (AL2) because they have more debugging options for AL2 compared to Ubuntu. However, AL2 comes with a very old kernel that doesn't support kind well (it's not possible to run two kind cluster at the same time at the same node). A lot of our tests/jobs are using kind, so that wasn't an option. Bottlerocket comes with a much newer kernel version (6.x), so we decided to give it a try instead.

Before that, we did a lot of debugging together with the AWS support team and the AWS EKS service team, but we were not able to come to a conclusion why Ubuntu nodes are failing. We're still unsure, but because chasing that would take a lot of time, we agreed it's not worth and we're going to stay on Bottlerocket.

However, we need to fix one more issue before declaring this cluster as "stable" and that's tracked by https://github.com/kubernetes/test-infra/issues/31135

I want to take a moment to thank to everyone from AWS for being so helpful and helping us solve this issue!!! Especially @tzneal and support folks!

/close
